tests/torch/test_sota_checkpoints.py

```diff
@@ -9,684 +9,535 @@
+            "Diff FP32": self.diff_fp32,
+            "Diff target": self.diff_target,
+            "Status": self.status,
+            "Build url": os.environ.get("BUILD_URL", ""),
+        }
+
+
+TEST_RESULT: List[ResultInfo] = []
+
+
+def add_test_result(run_info: ResultInfo):
+    TEST_RESULT.append(run_info)
+
+
+def read_reference_file(ref_path: Path) -> List[EvalRunParamsStruct]:
+    with ref_path.open(encoding="UTF-8") as source:
+        sota_eval_config = json.load(source, object_pairs_hook=OrderedDict)
 
-
-@pytest.mark.nightly
-class TestSotaCheckpoints:
     param_list = []
-    train_param_list = []
-    ids_list = []
-    train_ids_list = []
-    row_dict = OrderedDict()
-    color_dict = OrderedDict()
-    ref_fp32_dict = OrderedDict()
-    test = None
-
-    @staticmethod
-    def get_metric_file_name(model_name: str):
-        return "{}.metrics.json".format(model_name)
+    for sample_type_ in sota_eval_config:
+        datasets = sota_eval_config[sample_type_]
+        for dataset_name in datasets:
+            model_dict = datasets[dataset_name]
+            for model_name, sample_dict in model_dict.items():
+                if "target_pt" not in sample_dict:
+                    continue
+                param_list.append(
+                    EvalRunParamsStruct(
+                        model_name=model_name,
+                        config_name=sample_dict["config"],
+                        reference=sample_dict.get("reference", None),
+                        target_pt=sample_dict["target_pt"],
+                        target_ov=sample_dict["target_ov"],
+                        metric_type=sample_dict["metric_type"],
+                        dataset_name=dataset_name,
+                        sample_type=sample_type_,
+                        resume_file=sample_dict.get("resume", None),
+                        batch=sample_dict.get("batch", None),
+                        diff_fp32_min=sample_dict.get("diff_fp32_min", DIFF_FP32_MIN_GLOBAL),
+                        diff_fp32_max=sample_dict.get("diff_fp32_max", DIFF_FP32_MAX_GLOBAL),
+                        diff_target_ov_min=sample_dict.get("diff_target_ov_min", DIFF_TARGET_OV_MIN),
+                        diff_target_ov_max=sample_dict.get("diff_target_ov_max", DIFF_TARGET_OV_MAX),
+                        diff_target_pt_min=sample_dict.get("diff_target_pt_min", DIFF_TARGET_PT_MIN),
+                        diff_target_pt_max=sample_dict.get("diff_target_pt_max", DIFF_TARGET_PT_MAX),
+                        multiprocessing_distributed=sample_dict.get("multiprocessing_distributed", False),
+                        skip_ov=sample_dict.get("skip_ov", ""),
+                    )
+                )
+    return param_list
+
+
+EVAL_TEST_STRUCT = read_reference_file(Path(TEST_ROOT) / "torch" / "sota_checkpoints_eval.json")
+
+
+def idfn(val):
+    if isinstance(val, EvalRunParamsStruct):
+        return val.model_name
+    return None
```

Comment:
Please provide example of bad params about you tell.  
It's not clear how adding `NotImplementedError` can help with tests with same ID.
And how you can get same ID for multiple tests?

Reply:
If you worry about that you can add test case in `sota_checkpoints_eval.json` with not unique model name, it's possible, added check in `read_reference_file`.

Reply:
Looks like you want to be handling the case when the idfn can accept something other than `EvalRunParamsStruct`. I think you want to say that this situation is invalid, and so you return `None` as an ID. This leads to no errors in runtime, the tests will still be run. Instead you should make the situation fail even before the tests are started, so that no one mistakenly adds a non-`EvalRunParamsStruct` object to the list of parameters that this `idfn` is related to.

Reply:
To be clear, your comment confused me a bit, because as I know:

- pytest does not allow duplicate test ids by automatically adds an index to duplicate ids from idfn;
- if `idfn` returns None, will be used the default pytest converter to test id.

I am curious to know how it is possible to achieve the behavior you described.

Reply:
I didn't know that - I thought that returning `None` would generate a test ID of `"None"`, I've been seeing some of the test IDs with `"None"` but I guess they ended up in the test ID due to some other mechanism. So your current code will indeed pose no problems with duplicate IDs.

However, do you really want to do the `isinstance` check and allow non-`EvalRunParamsStruct` objects to be parameters for your test case? If a non-`EvalRunParamsStruct` is passed as a parameter, then this is most likely an error on part of the developer who edited the params list. Maybe it would be better to set up a type hint in the `idfn` signature and let the ID generation fail with an "AttributeError" if someone mistakenly puts a wrong object into test case params.

Reply:
test_id collect `None` if parameter is `None`, it is expected.

It's default template of idfn that provided in https://docs.pytest.org/en/7.4.x/example/parametrize.html#different-options-for-test-ids. And it's used in other NNCF tests.
If somebody will pass to incorrect data to fixture, test will fall anyway. Not `EvalRunParamsStruct` parameters can be used for idfn, and idfn can be extended to types.

Reply:
Looking out for more types than `EvalRunParamStruct` is probably overengineering here. But now that you've explained your intent, the solution makes sense.